### PR TITLE
Fix storage desync issue

### LINKED ISSUE #343: Storage bug
image

This issue is caused by initialising the transaction book with sample data when it is empty **without** considering the status of address book

### PR DESCRIPTION
**Pull request by seanlim**
Closes #343

### CODE REVIEW

File: src/main/java/spleetwaise/commons/MainApp.java [hunk 1]
```diff
@@ -91,51 +90,38 @@ public void init() throws Exception {
...
-        logger.info("Using data file : " + storage.getTransactionBookFilePath());
-
-        Optional<ReadOnlyTransactionBook> txnBookOptional;
-        ReadOnlyTransactionBook initialData;
-        try {
-            txnBookOptional = storage.readTransactionBook(addressBookModel);
-            if (!txnBookOptional.isPresent()) {
                 logger.info("Creating a new data file " + storage.getTransactionBookFilePath()
                         + " populated with a sample TransactionBook.");
+                addressBookModel = new AddressBookModelManager(SampleDataUtil.getSampleAddressBook());
```
Comment (SeeYangZhi) [inline src/main/java/spleetwaise/commons/MainApp.java:105]: Nit, should we move line 103 and line 104 to after line 105?
```suggestion
                addressBookModel = new AddressBookModelManager(SampleDataUtil.getSampleAddressBook());
                logger.info("Creating a new data file " + storage.getTransactionBookFilePath()
                        + " populated with a sample TransactionBook.");
```

File: src/main/java/spleetwaise/commons/MainApp.java [hunk 2]
```diff
@@ -91,51 +90,38 @@ public void init() throws Exception {
...
+                // Since addressbook is nonexistent, whatever state transactionbook is in should become irrelevant.
+                // We shall therefore start from sample data with transactionbook as well.
+                transactionBookModel = new TransactionBookModelManager(SampleDataUtil.getSampleTransactionBook());
+                return;
             }
-            initialData = txnBookOptional.orElseGet(SampleDataUtil::getSampleTransactionBook);
+
+            // Read the address book
+            addressBookModel = new AddressBookModelManager(addressBookOptional.get());
```
Comment (SeeYangZhi) [inline src/main/java/spleetwaise/commons/MainApp.java:113]: Should we add logs for these while we are at it

File: src/main/java/spleetwaise/commons/MainApp.java (hunk 1 above)
Comment (seanlim) [reply src/main/java/spleetwaise/commons/MainApp.java:104]: Hmm why though? the log message is to inform that there is no address book present

**Review by SeeYangZhi — APPROVED** [review]
Nice work, could have done some tests though!

**Pull request merged**